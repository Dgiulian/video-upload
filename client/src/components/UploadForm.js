import React, { useState } from 'react';

export default function UploadForm({ onUploadFiles }) {
  const [isDraggedOver, setIsDraggedOver] = useState(false);
  const handleFileSelect = (e) => {
    onUploadFiles(e.target.files);
  };
  const handleDragOver = (e) => {
    e.preventDefault();
    setIsDraggedOver(true);
  };
  const handleDragLeave = (e) => {
    e.preventDefault();

    setIsDraggedOver(false);
  };
  const handleDrop = (e) => {
    e.preventDefault();
    setIsDraggedOver(false);
    const files = Array.from(e.dataTransfer.files).filter((file) =>
      file.type.startsWith('video/')
    );
    onUploadFiles(files);
  };
  return (
    <form>
      <div>
        <div className="mt-6 grid grid-cols-1 row-gap-6 col-gap-4 sm:grid-cols-6">
          <div className="sm:col-span-6">
            <h2 className="leading-6 font-medium text-gray-700 text-lg">
              Upload Videos
            </h2>
            <div
              className={`mt-2 flex justify-center px-6 pt-5 pb-6 border-2 border-dashed rounded-md ${
                isDraggedOver ? 'border-blue-300' : 'border-gray-300'
              }`}
              onDragOver={handleDragOver}
              onDragLeave={handleDragLeave}
              onDrop={handleDrop}
              data-testid="drop-files"
            >
              <div className="text-center">
                <svg
                  className={`mx-auto h-12 w-12 ${
                    isDraggedOver ? 'text-blue-300' : 'text-gray-300'
                  }`}
                  stroke="currentColor"
                  fill="none"
                  viewBox="0 0 48 48"
                >
                  <path
                    d="M28 8H12a4 4 0 00-4 4v20m32-12v8m0 0v8a4 4 0 01-4 4H12a4 4 0 01-4-4v-4m32-4l-3.172-3.172a4 4 0 00-5.656 0L28 28M8 32l9.172-9.172a4 4 0 015.656 0L28 28m0 0l4 4m4-24h8m-4-4v8m-12 4h.02"
                    strokeWidth="2"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                  />
                </svg>
                <p className="mt-1 text-sm text-gray-600">
                  <label
                    type="button"
                    className="font-medium text-indigo-600 hover:text-indigo-500 focus:outline-none focus:underline transition duration-150 ease-in-out mr-1 cursor-pointer "
                  >
                    Upload a video
                    <input
                      type="file"
                      multiple={true}
                      className="hidden"
                      onChange={handleFileSelect}
                      accept="video/*"
                      name="videos"
                    />
                  </label>
                  or drag and drop
                </p>
                {/* <p className="mt-1 text-xs text-gray-500">
                  Mp4, Flv, GIF up to 10MB
                </p> */}
              </div>
            </div>
          </div>
        </div>
      </div>
      {/*       <div className="mt-8 border-t border-gray-200 pt-5">
        <div className="flex justify-end">
          <span className="inline-flex rounded-md shadow-sm">
            <Link
              to="/"
              type="button"
              className="py-2 px-4 border border-gray-300 rounded-md text-sm leading-5 font-medium text-gray-700 hover:text-gray-500 focus:outline-none focus:border-blue-300 focus:shadow-outline-blue active:bg-gray-50 active:text-gray-800 transition duration-150 ease-in-out"
            >
              Cancel
            </Link>
          </span>
          <span className="ml-3 inline-flex rounded-md shadow-sm">
            <button
              type="submit"
              className="inline-flex justify-center py-2 px-4 border border-transparent text-sm leading-5 font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-500 focus:outline-none focus:border-indigo-700 focus:shadow-outline-indigo active:bg-indigo-700 transition duration-150 ease-in-out"
            >
              Save
            </button>
          </span>
        </div>
      </div> */}
    </form>
  );
}
