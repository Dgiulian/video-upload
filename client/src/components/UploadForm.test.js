import React from 'react';
import { render, cleanup, fireEvent, act } from '@testing-library/react';
import UploadForm from './UploadForm';

describe('UploadForm', () => {
  afterEach(cleanup);

  test('It matches snapshot', () => {
    const wrapper = render(<UploadForm />);
    expect(wrapper).toMatchSnapshot();
  });

  test('onUploadFiles is called when files are selected', () => {
    const onUploadFiles = jest.fn();
    const { getByLabelText } = render(
      <UploadForm onUploadFiles={onUploadFiles} />
    );
    const inputEl = getByLabelText('Upload a video');
    fireEvent.change(inputEl);
    expect(onUploadFiles).toHaveBeenCalled();
  });

  test('onUploadFiles is called when files are dropped', () => {
    const onUploadFiles = jest.fn();
    const { getByTestId, debug } = render(
      <UploadForm onUploadFiles={onUploadFiles} />
    );
    const dropEl = getByTestId('drop-files');
    const fakeFiles = [
      {
        title: 'file1.mp4',
        type: 'video/mp4',
      },
      {
        title: 'file2.mp4',
        type: 'plain/txt',
      },
      {
        title: 'file3.mp4',
        type: 'video/flv',
      },
    ];
    const filteredFiles = fakeFiles.filter((file) =>
      file.type.startsWith('video')
    );
    fireEvent.drop(dropEl, { dataTransfer: { files: fakeFiles } });
    expect(onUploadFiles).toHaveBeenCalledWith(filteredFiles);
  });
});
