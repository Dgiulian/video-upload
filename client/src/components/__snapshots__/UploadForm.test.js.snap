// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadForm It matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div>
          <div
            class="mt-6 grid grid-cols-1 row-gap-6 col-gap-4 sm:grid-cols-6"
          >
            <div
              class="sm:col-span-6"
            >
              <h2
                class="leading-6 font-medium text-gray-700 text-lg"
              >
                Upload Videos
              </h2>
              <div
                class="mt-2 flex justify-center px-6 pt-5 pb-6 border-2 border-dashed rounded-md border-gray-300"
                data-testid="drop-files"
              >
                <div
                  class="text-center"
                >
                  <svg
                    class="mx-auto h-12 w-12 text-gray-300"
                    fill="none"
                    stroke="currentColor"
                    viewBox="0 0 48 48"
                  >
                    <path
                      d="M28 8H12a4 4 0 00-4 4v20m32-12v8m0 0v8a4 4 0 01-4 4H12a4 4 0 01-4-4v-4m32-4l-3.172-3.172a4 4 0 00-5.656 0L28 28M8 32l9.172-9.172a4 4 0 015.656 0L28 28m0 0l4 4m4-24h8m-4-4v8m-12 4h.02"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                  <p
                    class="mt-1 text-sm text-gray-600"
                  >
                    <label
                      class="font-medium text-indigo-600 hover:text-indigo-500 focus:outline-none focus:underline transition duration-150 ease-in-out mr-1 cursor-pointer "
                      type="button"
                    >
                      Upload a video
                      <input
                        accept="video/*"
                        class="hidden"
                        multiple=""
                        name="videos"
                        type="file"
                      />
                    </label>
                    or drag and drop
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div>
        <div
          class="mt-6 grid grid-cols-1 row-gap-6 col-gap-4 sm:grid-cols-6"
        >
          <div
            class="sm:col-span-6"
          >
            <h2
              class="leading-6 font-medium text-gray-700 text-lg"
            >
              Upload Videos
            </h2>
            <div
              class="mt-2 flex justify-center px-6 pt-5 pb-6 border-2 border-dashed rounded-md border-gray-300"
              data-testid="drop-files"
            >
              <div
                class="text-center"
              >
                <svg
                  class="mx-auto h-12 w-12 text-gray-300"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 48 48"
                >
                  <path
                    d="M28 8H12a4 4 0 00-4 4v20m32-12v8m0 0v8a4 4 0 01-4 4H12a4 4 0 01-4-4v-4m32-4l-3.172-3.172a4 4 0 00-5.656 0L28 28M8 32l9.172-9.172a4 4 0 015.656 0L28 28m0 0l4 4m4-24h8m-4-4v8m-12 4h.02"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
                <p
                  class="mt-1 text-sm text-gray-600"
                >
                  <label
                    class="font-medium text-indigo-600 hover:text-indigo-500 focus:outline-none focus:underline transition duration-150 ease-in-out mr-1 cursor-pointer "
                    type="button"
                  >
                    Upload a video
                    <input
                      accept="video/*"
                      class="hidden"
                      multiple=""
                      name="videos"
                      type="file"
                    />
                  </label>
                  or drag and drop
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
